crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -179,87 +179,97 @@ impl Inner {
 
         {
             // temporary to debug and compare with current implementation
-            let settlement = domain::settlement::Settlement::new(
-                transaction,
-                domain_separator,
-                &self.persistence,
-            )
-            .await;
-
-            // automatic checks vs current implementation
-            match (settlement, auction_data) {
-                (Ok(_), None) => {
-                    // bug: we should have an auction_data
-                    tracing::warn!(?auction_id, "automatic check error: missing auction_data");
+            let transaction = domain::settlement::Transaction::new(&transaction, domain_separator);
```

Comment:
Are we here still using the old code and just execute a bunch of tests to see if the new code behaves any different?
Could this be moved into a separate function to not clutter the business logic too much?
Also is this not testable via unit tests? Given that this is "just" decoding some calldata having unit tests for this doesn't sound too involved. 🤔

Reply:
> Are we here still using the old code and just execute a bunch of tests to see if the new code behaves any different?

Yes.

> Could this be moved into a separate function to not clutter the business logic too much?

Moved to a separate function.

> Also is this not testable via unit tests? Given that this is "just" decoding some calldata having unit tests for this doesn't sound too involved. 

We have a few unit tests for `domain::Settlement` building and calculating surplus and fees for it. It's a very exhaustive list of different cases that can happen even for valid settlements: solutions with and without protocol fees, solutions with multiple fees, solutions with cow amm orders and without, different order kinds (buy/sell), partially filled orders etc.

Testing this way give me far more confidence we won't break something in prod.